### PR TITLE
*: remove file sources

### LINKED ISSUE #11875: Remove non-cloud sources and sinks
This issue tracks the removal of sources and sinks that can't be used in Materialize Cloud. The work items are:

 * Removal of Avro OCF sinks
   * [x] Pivot dbt-materialize test suite to use a Kafka sink (it just needs _a_ sink)
   * [x] Remove sink parsing, planning, rendering and tests
 * Removal of Avro OCF sources
   * [x] Remove source parsing, planning, rendering, and tests
 * Removal of file sources
   * [x] Pivot dbt-materialize test suite to use a Kafka source (again, it just needs _a_ source)
   * [x] Pivot non-file-source specific tests to use S3 sources (this keeps the code paths as identical as possible), ideally over several PRs
   * [x] Remove parsing, planning, rendering, and file source-specific tests
 * [x] Remove safe mode (`--safe` CLI option)

I've tried to roughly break down the work into small, individually-PRable chunks. I'd like to avoid doing this in a single mega PR, as that will make it difficult to review whether we've preserved the appropriate tests.

### PR DESCRIPTION
**Pull request by benesch**
File sources don't work in Materialize Cloud, so remove them.

This commit also removes safe mode (`materialized --safe`), since safe
mode existed only to disallow usage of file sources.

Fixes #11875.
Closes #11936.

Co-authored-by: Gus Wynn <gus@materialize.com>

### Motivation

* This PR removes an obsolete feature.

### Testing

- [x] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

File: src/sql/src/pure.rs [hunk 1]
```diff
@@ -335,10 +293,19 @@ async fn purify_source_format_single(
...
-            delimiter,
+            delimiter: _,
             ref mut columns,
         } => {
-            purify_csv(file, connector, *delimiter, columns).await?;
+            if let CsvColumns::Header { names } = columns {
+                match connector {
+                    CreateSourceConnector::S3 { .. } => {
+                        if names.is_empty() {
+                            bail!("CSV WITH HEADER for S3 sources requires specifying the header columns");
```
Comment (guswynn) [inline src/sql/src/pure.rs:303]: nice! the railroad diagram needs to be adjusted to remove the skip over the header columns: https://materialize.com/docs/sql/create-source/s3/

removed here: https://github.com/MaterializeInc/materialize/blob/main/doc/user/layouts/partials/create-source/format/csv/syntax.html#L1 and fixed here https://github.com/MaterializeInc/materialize/blob/main/doc/user/sql-grammar/sql-grammar.bnf#L182-L193

**Review by guswynn — APPROVED** [review]
accepting, but I believe there are 2 doc sites that need to be fixed before merging

**Comment by guswynn** [comment]
Also, for readers of this pr, note that we are leaving the ocf-format reading code in src/avro/src/reader.rs so we can support avro ocf formatted values in s3 at some point

File: src/sql/src/pure.rs (hunk 1 above)
Comment (benesch) [reply src/sql/src/pure.rs:303]: Oh good point, thanks! Fixed.

**Pull request merged**